test/e2e/authorization.go

```diff
@@ -0,0 +1,572 @@
+					"persistentvolumes",
+					"persistentvolumeclaims",
+					"configmaps",
+				}
+				tc.data.verbs = []string{"create", "update", "delete", "patch"}
+			})
+			// This should be covered by the admission-controller tests.
+			// It's written here for completeness.
+			g.It("should deny write access in kube-system namespace", func() {})
+			g.It("should allow write access in namespaces other than kube-system", func() {
+				tc.data.namespaces = []string{"", "teapot"}
+				tc.run(context.TODO(), cs, true)
+				gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+			})
+		})
+
+		g.When("the resource is a global resource", func() {
+			g.BeforeEach(func() {
+				tc.data.verbs = []string{"create", "update", "delete", "patch"}
+			})
+			g.It("should deny access to Nodes", func() {
+				tc.data.resources = []string{"nodes"}
+				tc.run(context.TODO(), cs, false)
+				gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+			})
+			g.It("should allow access to resources other than Nodes", func() {
+				tc.data.resources = []string{
+					"namespaces",
+					"storage.k8s.io/storageclasses",
+					"apiextensions.k8s.io/customresourcedefinitions",
+				}
+				tc.run(context.TODO(), cs, true)
+				gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+			})
+		})
+	})
+
+	g.Context("For system users", func() {
+		var tc testCase
+
+		g.When("the user is kubelet", func() {
+			g.BeforeEach(func() {
+				tc.data.groups = [][]string{{"system:masters"}}
+				tc.data.users = []string{"kubelet"}
+			})
+			g.It("should allow to get Pods", func() {
+				tc.data.resources = []string{"pods"}
+				tc.data.verbs = []string{"get"}
+				tc.data.namespaces = []string{"teapot"}
+				tc.run(context.TODO(), cs, true)
+				gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+			})
+		})
+
+		g.When("the service account is daemonset-controller", func() {
+			g.BeforeEach(func() {
+				tc.data.users = []string{"system:serviceaccount:kube-system:daemon-set-controller"}
+				tc.data.groups = [][]string{{"system:serviceaccounts:kube-system"}}
+			})
+			g.It("should allow to update DaemonSet status subresource", func() {
+				tc.data.resources = []string{"apps/daemonsets/status"}
+				tc.data.verbs = []string{"update"}
+				tc.run(context.TODO(), cs, true)
+				gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+			})
+			g.It("should allow to update DaemonSet finalizers", func() {
+				tc.data.resources = []string{"apps/daemonsets/finalizers"}
+				tc.data.verbs = []string{"update"}
+				tc.run(context.TODO(), cs, true)
+				gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+			})
+			// TODO: Need to verify in the original tests if this is a permission on
+			// the controller-manager or the daemonset-controller.
+			// g.It("should allow to create Pods", func() {})
+		})
+
+		g.When("the default service account is in default namespace", func() {
```

Comment:
```suggestion
		g.When("the default service account in the default namespace", func() {
```

Reply:
This was taken care of when these two test cases were combined in: https://github.com/zalando-incubator/kubernetes-on-aws/pull/8536/commits/4eba59b8cb63f1d56b7ef05fad86022bfe76ebae